Docker build fails

Updating the rust version in the Dockerfile fixes this.
```
#12 0.554   Installing rust-rtsp-server v0.1.0 (/usr/src/app)
#12 0.563     Updating crates.io index
#12 126.9  Downloading crates ...
#12 127.5   Downloaded failure v0.1.8
#12 127.5   Downloaded uuid v0.8.2
#12 127.6   Downloaded futures v0.3.21
#12 127.7   Downloaded chrono v0.4.19
#12 127.8   Downloaded clap v2.34.0
#12 127.9   Downloaded backtrace v0.3.64
#12 128.0   Downloaded serde v1.0.136
#12 128.1   Downloaded getrandom v0.2.4
#12 128.1   Downloaded failure_derive v0.1.8
#12 128.4   Downloaded atty v0.2.14
#12 128.4   Downloaded strsim v0.8.0
#12 128.4   Downloaded unicode-width v0.1.9
#12 128.4   Downloaded miniz_oxide v0.4.4
#12 128.4   Downloaded rustc-demangle v0.1.21
#12 128.5   Downloaded cc v1.0.72
#12 128.5   Downloaded futures-channel v0.3.21
#12 128.5   Downloaded num-traits v0.2.14
#12 128.6   Downloaded futures-core v0.3.21
#12 128.6   Downloaded futures-task v0.3.21
#12 128.6   Downloaded futures-executor v0.3.21
#12 128.7   Downloaded futures-util v0.3.21
#12 128.8   Downloaded ansi_term v0.12.1
#12 128.8   Downloaded vec_map v0.8.2
#12 128.8   Downloaded bitflags v1.3.2
#12 128.8   Downloaded addr2line v0.17.0
#12 128.8   Downloaded futures-sink v0.3.21
#12 128.8   Downloaded time v0.1.43
#12 128.8   Downloaded num-integer v0.1.44
#12 129.0   Downloaded object v0.27.1
#12 129.0   Downloaded textwrap v0.11.0
#12 129.3   Downloaded cfg-if v1.0.0
#12 129.5   Downloaded futures-io v0.3.21
#12 129.5   Downloaded gstreamer v0.15.7
#12 129.6   Downloaded libc v0.2.117
#12 129.7   Downloaded quote v1.0.15
#12 129.7   Downloaded pin-utils v0.1.0
#12 129.7   Downloaded pin-project-lite v0.2.8
#12 129.7   Downloaded synstructure v0.12.6
#12 129.7   Downloaded futures-macro v0.3.21
#12 129.7   Downloaded proc-macro2 v1.0.36
#12 129.7   Downloaded memchr v2.4.1
#12 129.8   Downloaded adler v1.0.2
#12 129.8   Downloaded autocfg v1.1.0
#12 129.9   Downloaded syn v1.0.86
#12 129.9   Downloaded slab v0.4.5
#12 130.4   Downloaded gimli v0.26.1
#12 130.5   Downloaded lazy_static v1.4.0
#12 130.5   Downloaded glib-sys v0.9.1
#12 130.6   Downloaded gobject-sys v0.9.1
#12 130.6   Downloaded cfg-if v0.1.10
#12 130.6   Downloaded unicode-xid v0.2.2
#12 130.6   Downloaded paste v0.1.18
#12 130.6   Downloaded muldiv v0.2.1
#12 130.6   Downloaded num-rational v0.2.4
#12 130.6   Downloaded glib v0.9.3
#12 130.6   Downloaded gstreamer-sys v0.8.1
#12 130.9   Downloaded pkg-config v0.3.24
#12 130.9   Downloaded proc-macro-hack v0.5.19
#12 131.0   Downloaded paste-impl v0.1.18
#12 131.0    Compiling autocfg v1.1.0
#12 131.1    Compiling libc v0.2.117
#12 133.8    Compiling proc-macro2 v1.0.36
#12 136.7    Compiling unicode-xid v0.2.2
#12 137.1    Compiling pkg-config v0.3.24
#12 137.3    Compiling syn v1.0.86
#12 139.7    Compiling memchr v2.4.1
#12 141.2    Compiling futures-core v0.3.21
#12 142.1    Compiling futures-channel v0.3.21
#12 142.9    Compiling futures-task v0.3.21
#12 144.2    Compiling futures-util v0.3.21
#12 145.2    Compiling futures-sink v0.3.21
#12 145.6    Compiling pin-project-lite v0.2.8
#12 145.8    Compiling pin-utils v0.1.0
#12 145.9    Compiling proc-macro-hack v0.5.19
#12 146.3    Compiling futures-io v0.3.21
#12 147.6    Compiling slab v0.4.5
#12 147.9    Compiling cfg-if v1.0.0
#12 148.1    Compiling cc v1.0.72
#12 148.2    Compiling adler v1.0.2
#12 149.0    Compiling serde v1.0.136
#12 151.3    Compiling gimli v0.26.1
#12 172.6    Compiling bitflags v1.3.2
#12 172.7    Compiling failure_derive v0.1.8
#12 174.7    Compiling unicode-width v0.1.9
#12 175.1    Compiling lazy_static v1.4.0
#12 175.2    Compiling rustc-demangle v0.1.21
#12 175.3    Compiling gstreamer v0.15.7
#12 175.3 error[E0658]: use of unstable library feature 'matches_macro'
#12 175.3    --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/rustc-demangle-0.1.21/src/v0.rs:700:20
#12 175.3     |
#12 175.3 700 |                 if matches!((quote, c), ('\'', '"') | ('"', '\'')) {
#12 175.3     |                    ^^^^^^^
#12 175.3     |
#12 175.3     = note: for more information, see https://github.com/rust-lang/rust/issues/65721
#12 175.3
#12 175.8    Compiling ansi_term v0.12.1
#12 176.3 error: aborting due to previous error
#12 176.3
#12 176.3 For more information about this error, try `rustc --explain E0658`.
#12 176.3 error: could not compile `rustc-demangle`.
#12 176.3 warning: build failed, waiting for other jobs to finish...
#12 177.8 error: failed to compile `rust-rtsp-server v0.1.0 (/usr/src/app)`, intermediate artifacts can be found at `/usr/src/app/target`
#12 177.8
#12 177.8 Caused by:
#12 177.8   build failed
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -25,7 +25,7 @@ RUN apt-get install --no-install-recommends -y wget ca-certificates
 ENV RUSTUP_HOME=/usr/local/rustup \
     CARGO_HOME=/usr/local/cargo \
     PATH=/usr/local/cargo/bin:$PATH \
-    RUST_VERSION=1.40.0
+    RUST_VERSION=1.58.1
 
 RUN set -eux; \
     dpkgArch="$(dpkg --print-architecture)"; \

